### PR TITLE
Implement titles for users

### LINKED ISSUE #69: Make some titles
These should be displayed next to the user.

### PR DESCRIPTION
**Pull request by dwoos**
Work in progress for now

Also, this will close #69.

### CODE REVIEW

**Comment by mwillsey** [comment]
What's left on this?

Also, this will close #69.

**Comment by emichael** [comment]
Equipping and unequipping titles seems to work now.

**Comment by emichael** [comment]
AFAICT now, the only thing left to do is open up the inventory page to everyone and then give people a link to it (probably when they get a new title).

File: crossbot/slack/commands/add.py [hunk 1]
```diff
@@ -58,7 +58,9 @@ def add(request):
         text = "*{} Added*: {}".format(time.SHORT_NAME, time.time_str())
         if args.date != parse_date('now'):
             text += " {}".format(time.date)
-        response.add_text(text, ephemeral=False)
+        response.attach(
+            text=text, author_name=request.user.title_text, ephemeral=False
+        )
```
Comment (mwillsey) [inline crossbot/slack/commands/add.py]: We need to have `"mrkdwn_in": ["text"]` in the json. See [here](https://api.slack.com/docs/messages/builder?msg=%7B%22attachments%22%3A%5B%7B%22author_name%22%3A%22My%20cool%20title%20name%22%2C%22text%22%3A%22*Mini%20added*%3A%200%3A10%22%2C%22mrkdwn_in%22%3A%5B%22text%22%5D%7D%5D%7D)

File: crossbot/models.py [hunk 2]
```diff
@@ -385,6 +420,26 @@ def streaks(entries):
 
         return streaks
 
+    @classmethod
```
Comment (mwillsey) [inline crossbot/models.py]: This needs a test.

File: crossbot/models.py [hunk 3]
```diff
@@ -38,7 +38,8 @@ class Meta:
     image_url = models.CharField(max_length=150, blank=True)
 
     crossbucks = models.IntegerField(default=0)
-    hat_key = models.CharField(max_length=20, null=True, blank=True)
+    hat_key = models.CharField(max_length=40, null=True, blank=True)
+    title_key = models.CharField(max_length=40, null=True, blank=True)
```
[the remark this answers is not in the document]
Comment (emichael) [reply crossbot/models.py]: `null` is a db constraint. `blank` means the field can be blank in generated forms (i.e., in the admin panel). We want both.

File: crossbot/settings.py [hunk 4]
```diff
@@ -3,3 +3,4 @@
 # Placing crossbot settings in here for now
 CROSSBUCKS_PER_SOLVE = getattr(s, 'CROSSBOT_CROSSBUCKS_PER_SOLVE', 10)
 ITEM_DROP_RATE = getattr(s, 'CROSSBOT_ITEM_DROP_RATE', 0.1)
+DEFAULT_TITLE = getattr(s, 'CROSSBOT_DEFAULT_TITLE', "Crossworder")
```
[the remark this answers is not in the document]
Comment (emichael) [reply crossbot/settings.py]: I disagree. I thought we already talked about "Crossworder" being the default title?

File: crossbot/slack/commands/add.py (hunk 1 above)
Comment (emichael) [reply crossbot/slack/commands/add.py]: Weird. It worked for me before without `mrkdown_in`.

File: crossbot/settings.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (mwillsey) [reply crossbot/settings.py]: Hm, so it seems you do need a author name to show an image. Ok im fine with this.

**Pull request merged**